problem_builder/mentoring.py

```diff
@@ -776,3 +789,98 @@ def workbench_scenarios():
+
+    def student_view(self, context):
+        fragment = Fragment()
+        child_content = u""
+
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if child is None:  # child should not be None but it can happen due to bugs or permission issues
+                child_content += u"<p>[{}]</p>".format(self._(u"Error: Unable to load child component."))
+            elif not isinstance(child, MentoringMessageBlock):
+                try:
+                    child_fragment = child.render('mentoring_view', context)
+                except NoSuchViewError:
+                    if child.scope_ids.block_type == 'html' and getattr(self.runtime, 'is_author_mode', False):
+                        # html block doesn't support mentoring_view, and if we use student_view Studio will wrap
+                        # it in HTML that we don't want in the preview. So just render its HTML directly:
+                        child_fragment = Fragment(child.data)
+                    else:
+                        child_fragment = child.render('student_view', context)
+                fragment.add_frag_resources(child_fragment)
+                child_content += child_fragment.content
+
+        fragment.add_content(loader.render_template('templates/html/mentoring.html', {
+            'self': self,
+            'title': self.display_name,
+            'show_title': self.show_title,
+            'child_content': child_content,
+        }))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/problem-builder.css'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/vendor/underscore-min.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/util.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/mentoring_standard_view.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/mentoring.js'))
+        fragment.add_resource(loader.load_unicode('templates/html/mentoring_attempts.html'), "text/html")
+        fragment.add_resource(loader.load_unicode('templates/html/mentoring_grade.html'), "text/html")
+        fragment.add_resource(loader.load_unicode('templates/html/mentoring_review_questions.html'), "text/html")
+
+        self.include_theme_files(fragment)
+        # Workbench doesn't have font awesome, so add it:
+        if WorkbenchRuntime and isinstance(self.runtime, WorkbenchRuntime):
+            fragment.add_css_url('//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css')
+
+        fragment.initialize_js('MentoringBlock')
+
+        return fragment
+
+    def author_edit_view(self, context):
+        """
+        Add some HTML to the author view that allows authors to add child blocks.
+        """
+        fragment = Fragment(u'<div class="mentoring">')  # This DIV is needed for CSS to apply to the previews
```

Comment:
@itsjeyd please use StudioContainerWithNestedXBlocksMixin - it would look something like that:

    context['wrap_children'] = {
         'head': u'<div class="mentoring">',
         'tail': u'</div>'
    }
    fragment = super(MentoringWithExplicitStepsBlock, self).author_edit_view(context)
    fragment.add_content(loader.render_template('templates/html/mentoring_url_name.html', {
            "url_name": self.url_name
     }))
     # css and js stuff

`mentoring_edit.js` is mostly not needed, except for `ProblemBuilderUtil.transformClarifications(element);` Not sure what does it do - @bradenmacdonald - some hint on that? Anyway, if you still need it, it would be better to have a new `mentoring_with_steps_edit.js` with the following contents:

    function MentoringWithStepsEdit(runtime, element) {
        ProblemBuilderUtil.transformClarifications(element);
        StudioEditableXBlockMixin(runtime, element);
    }